src/main/java/ml/shifu/shifu/util/CommonUtils.java

```diff
@@ -248,7 +249,40 @@ public static ModelConfig loadModelConfig() throws IOException {
      *             if {@code path} is null or empty, if sourceType is null.
      */
     public static ModelConfig loadModelConfig(String path, SourceType sourceType) throws IOException {
-        return loadJSON(path, sourceType, ModelConfig.class);
+        ModelConfig modelConfig = loadJSON(path, sourceType, ModelConfig.class);
+        if(modelConfig.getTrain().getGridConfigFile() != null) {
+            // Only load file content. Grid search params parsing is done in {@link GridSearch} initialization.
+            modelConfig.getTrain().setGridConfigFileContent(loadFileContent(modelConfig.getTrain().getGridConfigFile(), sourceType));
```

Comment:
One question, if configuration file is in local, how to make sure such file can also be read in HDFS mode. Like in NNMaster, it is run in cluster, can you read such configuration file from HDFS?

Reply:
The code reads file with sourcetype using ShifuFileUtils.getReader(path, sourceType), same as reading ModelConfig.json. I suppose that would handle file location.
I will add code to copy gridsearch config file from local to HDFS to make sure it is at the same   location as ModelConfig.json.

Reply:
Please check latest commits. gridsearch config file will be at the same folder with ModelConfig.json if in HDFS.